src/scope/async_hooks/async_wrap.js

```diff
@@ -10,28 +10,56 @@ try {
   asyncHook = require('async-hook-jl')
```

Comment:
I am not sure I understand the relationship between `async_hooks`, `async-hook`, and `async-hook-jl`, and how we are using them.

Also, if `async-hook-jl` is more recent, shouldn't we try to load that one first ?

Reply:
This existed before this PR and wasn't changed.

Reply:
yeah, I know, I just don't know much about it.

Reply:
> I am not sure I understand the relationship between `async_hooks`, `async-hook`, and `async-hook-jl`, and how we are using them.

Basically `async_hooks` is the (hopefully) final solution in Node for keeping track of asynchronous context since Node 8.

There was also a previous undocumented attempt as a C++ API called `AsyncWrap` that was available through `process.binding('async_wrap')`. The `async-hook` and `async-hook-jl` modules are simple JS wrappers on top of it that also solve some of its limitations such as the inability to manage multiple hooks. `async-hook-js` was forked from `async-hook` and never merged back, but it's used in `cls-hooked` which is a popular alternative to `continuation-local-storage` since it supports async/await.

> Also, if `async-hook-jl` is more recent, shouldn't we try to load that one first ?

That is a valid point but I'm not sure if it's in the scope of this PR. I'm willing to change it if you think it's warranted.

Reply:
Thanks, super helpful.

> That is a valid point but I'm not sure if it's in the scope of this PR. I'm willing to change it if you think it's warranted.

Agree not needed in this PR, your call on if we should address.

Reply:
In any case, using both at the same time would result in a lot of issues, so there is not much we can do there regardless of their priority.